src/game/collide.c

```diff
@@ -937,6 +961,93 @@ int32_t Move3DPosTo3DPos(
         && srcpos->y_rot == destpos->y_rot && srcpos->z_rot == destpos->z_rot;
 }
 
+int32_t Move3DPosTo3DPosAnim(
+    PHD_3DPOS *srcpos, PHD_3DPOS *destpos, int32_t velocity, int16_t rotation,
+    ITEM_INFO *lara_item)
+{
+    int32_t dx = destpos->x - srcpos->x;
+    int32_t dy = destpos->y - srcpos->y;
+    int32_t dz = destpos->z - srcpos->z;
+    int32_t distance = phd_sqrt(SQUARE(dx) + SQUARE(dy) + SQUARE(dz));
+    if (velocity >= distance) {
+        srcpos->x = destpos->x;
+        srcpos->y = destpos->y;
+        srcpos->z = destpos->z;
+    } else {
+        int32_t shift = PHD_ONE * velocity / distance;
+        srcpos->x += shift * dx >> 16;
+        srcpos->y += shift * dy >> 16;
+        srcpos->z += shift * dz >> 16;
+    }
+
+    if (!g_Lara.interact_target.is_moving) {
+        if (g_Lara.water_status != LWS_UNDERWATER) {
+            const int16_t stepToAninNum[4] = {
+                AA_SIDESTEP_LEFT,
+                AA_WALK_FORWARD,
+                AA_SIDESTEP_RIGHT,
+                AA_WALK_BACK,
+            };
+            const int16_t stepToAnimState[4] = {
+                AS_STEPLEFT,
+                AS_WALK,
+                AS_STEPRIGHT,
+                AS_BACK,
+            };
+
+            int32_t angle =
+                (PHD_ONE
+                 - phd_atan(srcpos->x - destpos->x, srcpos->z - destpos->z))
+                % PHD_ONE;
+            uint32_t quadrant =
+                ((((uint32_t)(angle + PHD_45) >> W2V_SHIFT)
+                  - ((uint16_t)(destpos->y_rot + PHD_45) >> W2V_SHIFT))
+                 & 0x3);
+
+            lara_item->anim_number = stepToAninNum[quadrant];
+            lara_item->frame_number =
+                g_Anims[lara_item->anim_number].frame_base;
+            lara_item->goal_anim_state = stepToAnimState[quadrant];
+            lara_item->current_anim_state = stepToAnimState[quadrant];
+
+            g_Lara.gun_status = LGS_HANDSBUSY;
+        }
+
+        g_Lara.interact_target.is_moving = 1;
+        g_Lara.interact_target.move_count = 0;
+    }
+
+    int16_t adiff = destpos->x_rot - srcpos->x_rot;
+    if (adiff > rotation)
```

Comment:
Missing `{`. Are you sure you've read the coding conventions? :)

Reply:
Lol whoops. Another TR5 thing.